src/data/set/finite.lean

```diff
@@ -310,6 +310,10 @@ instance fintype_prod (s : set α) (t : set β) [fintype s] [fintype t] :
   fintype (s ×ˢ t : set (α × β)) :=
 fintype.of_finset (s.to_finset ×ˢ t.to_finset) $ by simp
 
+@[simp] lemma to_finset_prod (s : set α) (t : set β) [fintype s] [fintype t] :
+  (s ×ˢ t).to_finset = s.to_finset ×ˢ t.to_finset :=
+by { ext, simp only [set.mem_to_finset, set.mem_prod, finset.mem_product] }
```

Comment:
I was thinking about this one more. I believe correct place to put it is where I put it in #16459. We can either wait for that to be merged (should be quick I hope) or you can copy what I have there and I'll delete that PR -- whatever you want to do!

Reply:
Let's wait for your PR then! But I'm confused:I believe I tried exactly that but got an error, which I understood to be because the `fintype` instance for the product is only defined at line 894. Wasn't that the case?

Reply:
Aha, nevermind, you added `fintype` as an assumption… neat!

Reply:
Yeah, it turns out these lemmas work better if everything that takes a `fintype` instance gets its own `fintype` argument. For example, with this lemma we want it to be able to apply to any `(s ×ˢ t).to_finset` term, even if its `fintype` instance isn't the standard one.

Also, #16459 has been merged!

Reply:
OK, I rebased and pushed. Maybe that was not the correct strategy, but anyway…

Reply:
It looks like this wasn't the correct strategy since GitHub thinks you've modified 143 files. I'll try to untangle what happened.